test/timer-test.ts

```diff
@@ -0,0 +1,190 @@
+            lastTaskId = taskId;
+        }
+    }
+    
+    let promise = new Promise<string>((resolve, reject) => {
+        let execResult = "";
+        for (let taskId = 0; taskId < kTaskGroupSize; taskId++) {
+            let wait = 300 * (taskGroupId * kTaskGroupSize + taskId + 1);
+            let startTime = Date.now();
+            let timeout = setTimeout((lastTaskId: number) => {
+                let waitToRun = Date.now() - startTime;
+                execResult = `${execResult}:${taskId}_OnTime`;
+                if (Math.abs(waitToRun - wait) > kAllowedScheduleDiffInMS) {
+                    execResult = `${execResult}(X)`;
+                }
+
+                if (taskId == lastTaskId) {
+                    if (execResult == correctResult) {
+                        resolve(`OK:${execResult}`)
+                    }
+                    else {
+                        reject(`FAIL:${execResult} .vs. ${correctResult}`)
+                    }
+                }
+            }, wait, lastTaskId);
+
+            if (taskId == 1) {
+                clearTimeout(timeout);
+            }
+        }
+    });
+
+    return promise;
+}
+
+export function setIntervalTest(taskGroupId: number, duration: number, count: number) : Promise<string> {
+    const kAllowedScheduleDiffInMS = 200;
+
+    let correctResult = "";
+    for (let i = 0; i < count; ++i) {
+        correctResult += `:${i}_OnTime`
+    }
+
+    let repeatCount = 0;
+    let execResult = "";
+    let startTime = Date.now();
+    let interval = setInterval(() => {
+        let wait = Date.now() - startTime;
+        execResult += `:${repeatCount}_OnTime`;
+        ++repeatCount;
+        let avgScheduleDiff = Math.abs(wait - repeatCount * duration) / repeatCount;
+        if (avgScheduleDiff > kAllowedScheduleDiffInMS) {
+            execResult += `(X)`;
+        }
+    }, duration);
+
+    let promise = new Promise<string>((resolve, reject) => {
+        setTimeout(() => {
+            if (execResult == correctResult) {
+                resolve(`OK:${execResult}`)
+            }
+            else {
+                reject(`FAIL:${execResult} .vs. ${correctResult}`)
+            }
+        }, duration * (count + 2.6));
+    });
+
+    setTimeout(()=> {
+        clearInterval(interval);
+    },  Math.ceil(duration * (count + 0.8)));
+
+    return promise;
+}
+
+
+declare var __in_napa: boolean;
+if (typeof __in_napa === 'undefined') {
+    let assert = require('assert');
+
+    const NUMBER_OF_WORKERS = 3;
+    const kTaskGroupCount = 3;
+    let zone = napa.zone.create('zone', { workers: NUMBER_OF_WORKERS });
+
+    describe("napajs/timers", () => {
+    
+        describe("setImmediate/clearImmediate", function() {
+            let promises: Promise<napa.zone.Result>[] = [];
+            for (let groupId = 0; groupId < kTaskGroupCount; groupId++) {
+                let res = zone.execute('./timer-test', 'setImmediateTest', [groupId]);
```

Comment:
create another file

Reply:
I thinks this is same.

Reply:
And it could let people notice some tricky part of current implementation.